### PR TITLE
fix(insights): Improve load more for breakdowns

### LINKED ISSUE #23554: Trends total value tables should be paginateable
https://posthoghelp.zendesk.com/agent/tickets/15364

> Pagination on tables appears to broken.
> 
> Viewing this insight (https://us.posthog.com/project/2/insights/bn1ojimP) shows around 20 entries, but when I exported the data is gave me around 60.
>
> There should be a 'Load more' option on the insight, but it's not appearing.

### LINKED ISSUE #24307: Checkbox for all rows on breakdown table isn't working outside edit mode
The top left checkbox isn't working when outside of edit mode.

Image

https://posthoghelp.zendesk.com/agent/tickets/16575

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

It was only working if you selected "display other", because presumably you otherwise couldn't tell if more breakdown values are available.

Fixes #23554

## Changes

- make it show up in all cases
- adjust the wording to have the actual values - since you also modify the query with this and it's persisted when you save
- use separate `hasMore` flag that is derived from "other" breakdowns being available. For this to work we ignore the `hide_other` flag and always calculate them, but hide the label+value later.

image

## How did you test this code?

Tested locally

### CODE REVIEW

File: frontend/__snapshots__/scenes-app-experiments--running-trend-experiment--dark.png
Comment (thmsobrmlr) [inline frontend/__snapshots__/scenes-app-experiments--running-trend-experiment--dark.png:1]: Seems this affects experiment results as well, where we don't need it.

File: frontend/__snapshots__/scenes-app-insights--trends-area-breakdown--dark--webkit.png
Comment (thmsobrmlr) [inline frontend/__snapshots__/scenes-app-insights--trends-area-breakdown--dark--webkit.png:1]: In general we should probably only show this if there are more breakdown values.

**Review by skoob13 — APPROVED** [review]

**Comment by webjunkie** [comment]
@thmsobrmlr I've added a relatively quick approach/workaround to tell if there are actually more values. Can you check again? Also for experiments, I checked one and now the load more isn't there. But maybe that depends on certain conditions, so would be great if you can also check that.

**Review by thmsobrmlr — CHANGES_REQUESTED** [review]
Thanks for the follow-up @webjunkie! Works great now.

Did a couple of stylistic changes, hope you don't mind.

There's only one thing blocking this from merging now: The button only appears on a new insight, but not after saving. I'm assuming this has something to do with the insight serializer not returning the `hasMore` or similar.

**Comment by thmsobrmlr** [comment]
> Also for experiments, I checked one and now the load more isn't there. But maybe that depends on certain conditions, so would be great if you can also check that.

This was just an artifact of the previous implementation where we would always display the button regardless if there are more values. Experiments only have two values and as such it's working there now.

**Comment by webjunkie** [comment]
> with the insight serializer not returning the hasMore or similar.

Okay fixed this now. The `hasMore` is next to the results... which means it needs to go into the serializer as well. This whole area is a bit cumbersome. I wonder why no other insights had specific fields that needed to be included. We have `hasMore` also for other query responses, but it seems it's mostly actors responses that are not going to be the result of a saved insight 🤔

**Comment by thmsobrmlr** [comment]
> > with the insight serializer not returning the hasMore or similar.
> 
> Okay fixed this now. The `hasMore` is next to the results... which means it needs to go into the serializer as well. This whole area is a bit cumbersome. I wonder why no other insights had specific fields that needed to be included. We have `hasMore` also for other query responses, but it seems it's mostly actors responses that are not going to be the result of a saved insight 🤔

Yeah it's quite annoying. We did have other insights with problems here and probably still have some edge cases that are unhandled..

**Review by thmsobrmlr — APPROVED** [review]

**Pull request merged**